### PR TITLE
Start Keploy server from Typescript-SDK

### LINKED ISSUE #352: [feature]: start Keploy server from Typescript-SDK
### Is there an existing feature request for this?

- [X] I have searched the existing issues

### Summary

Currently, we need to run Keploy binary and then run our application which is integrated with TS-SDK. 

 We can run the keploy binary from SDK and we don't have to see that whether the keploy binary is running or not.

### Why should this be worked on?

If Keploy binary is not running then,  error will be thrown on making API request on the application.

### PR DESCRIPTION
**Pull request by frey0-0**
Fixes [#352](https://github.com/keploy/keploy/issues/352)
If the keploy mode is record and test then it will check if keploy is running on port 6789 and if it isn't then it will start an asynchronously process to start the keploy server and the server won't keeping running, it will stop the server as soon as the main app server stops

### CODE REVIEW

**Comment by re-Tick** [comment]
Hi @Frey0-0, we should start the Keploy server from Keploy class constructor in src/keploy.ts because middleware is called on API call only. But we should check for Keploy binary during user API server setup.

**Comment by Frey0-0** [comment]
Yes that makes sense @re-Tick cause it may be possible the binary is not even present. I will look into this and mention if I have any issues here only. Thanks for the review 😄

**Comment by Frey0-0** [comment]
@re-Tick I have made the changes now we check if the `keploy` binary is present or not and next if it is running on port `6789` and if it is present and not running then it will spawn a process to start the server

**Comment by re-Tick** [comment]
Hi @Frey0-0, I have tried these changes in my local but the keploy binary do not run from the spawn method call.

**Comment by Frey0-0** [comment]
the spawn method call will run the binary as long as the node server is running and it will be terminated if the node server is stopped because the parent process has stopped. Please have a look at this video below. 

This is on my system which has the keploy binary installed
[Screen Recording](https://user-images.githubusercontent.com/94757729/224575122-a11247b6-6a81-4829-91fa-7ab246afb94d.webm)

This is a system which doesn't have the keploy binary
Screenshot from 2023-03-13 02-52-04

The keploy server stops if we stop the node server and can be started from the command again
[Screen Recording ](https://user-images.githubusercontent.com/94757729/224575543-f0e3e4bc-350f-4011-a046-96137e5579dd.webm)

**Comment by re-Tick** [comment]
I have found an interesting issue due to which Keploy binary is trying to run on the `same port` as the sample API server. This is because Keploy runs on port defined by `PORT` env variable or 6789 (default). 

So, we can solve this by changing the env variable name in [keploy](https://github.com/keploy/keploy/blob/ee0e9c33ebeedaf213713e03205343146e21449d/server/server.go#L66) to "KPORT" 

You can repklicate this by: 
1. Use node-fetch sample of samples/typescript.
2. Add `export PORT=8080` in the [.env](https://github.com/keploy/samples-typescript/blob/main/node-fetch/.env) file of node-fetch.
3. Now,  Change the env varibale name from `KEPLOY_APP_PORT` to `PORT` in [this](https://github.com/keploy/samples-typescript/blob/7a26737be3aa0f1f65a898fc0710095f1e34a3e1/node-fetch/server.js#L31) line of app.listen
4. Try recording some testcases. 
Error will be thrown `failed to post testcase to keploy server.`

**Comment by Frey0-0** [comment]
Oh yes! I faced the same error and we should change the name of the Keploy Port. Should I add an issue and then raise a PR for this or directly raise a PR for this describing the issue for the variable name change?

**Comment by Frey0-0** [comment]
I have created an issue and also linked a PR for it. Please review it :) and are there any more changes that need to be done on this PR?

File: src/keploy.ts [hunk 1]
```diff
@@ -76,6 +77,25 @@ export default class Keploy {
...
+      exec("which keploy", (error, stdout, stderr) => {
+        if (!stdout) {
+          console.log(
+            "Keploy Server is not installed. Please install it using https://docs.keploy.io/docs/go/installation "
+          );
+          process.exit();
+        } else {
+          exec("lsof -i:6789", (error, stdout, stderr) => {
+            if (!stdout) {
+              spawn("keploy");
```
Comment (re-Tick) [inline src/keploy.ts:93]: Here, run the Keploy binary with `KEPLOY_PORT=6789` env variable.

Comment (Frey0-0) [reply src/keploy.ts:93]: So I should use the command `export KEPLOY_PORT=6789` before spawning the keploy binary right?

Comment (re-Tick) [reply src/keploy.ts:93]: Yes, this export cmd can be run with keploy using ` && ` in the exec method

Comment (Frey0-0) [reply src/keploy.ts]: I have done the changes please have a look :sparkles:

Comment (re-Tick) [reply src/keploy.ts]: LGTM. Please rebase your commit messages for the github action to pass

Comment (Frey0-0) [reply src/keploy.ts]: I have made the changes to commit messages. I didn't know there was a specific way to write the commit messages. Thank you for being so patient :sweat_smile:

**Comment by re-Tick** [comment]
LGTM. Please add the link in GSOC task list

**Pull request closed without merging**